iaso/api/xls_form_generator.py

```diff
@@ -0,0 +1,257 @@
+                    ] = f"select_one ou_{str(k).lower()}"
+                    q_sheet[cell[get_column_position("name", q_sheet) - 1] + str(starting_row)] = f"ou_{str(k).lower()}"
+                    q_sheet[cell[get_column_position("label", q_sheet) - 1] + str(starting_row)] = f"Select {k}"
+                    q_sheet[cell[get_column_position("required", q_sheet) - 1] + str(starting_row)] = "yes"
+                    if ou_hierarchy_list.index(k.lower()) != 0:
+                        parent_type = ou_hierarchy_list[ou_hierarchy_list.index(str(k.lower())) - 1]
+                        q_sheet[cell[get_column_position("choice_filter", q_sheet) - 1] + str(starting_row)] = (
+                            f"{parent_type}=$" "{" f"ou_{parent_type}" "} "
+                        )
+                # Add org units as hierarchy into the choices sheet.
+                sheet[cell[choices_column - 1] + str(choices_row)] = f"ou_{str(k).lower()}"
+                sheet[cell[choices_column] + str(choices_row)] = v.id
+                sheet[cell[choices_column + 1] + str(choices_row)] = str(v.name)
+
+                index_hierarchy = ou_hierarchy_list.index(str(k.lower()))
+                ou_dic = {k.lower(): v for k, v in ou_dic.items()}
+
+                calculation_index = 0
+                if index_hierarchy > 0:
+                    index_hierarchy -= 1
+                    for row_calc in sheet.rows:
+                        iterator = 0
+                        for s_cell in row_calc:
+                            iterator += 1
+                            if str(s_cell.value).lower() == ou_hierarchy_list[index_hierarchy].lower():
+                                calculation_index += iterator
+                                break
+                parent_ou_id = ou_dic.get(ou_hierarchy_list[index_hierarchy].lower()).id
+                sheet[cell[choices_column + calculation_index - 2] + str(choices_row)] = str(parent_ou_id)
+
+                choices_row += 1
+                starting_row += 1
+                survey_last_empty_row += 1
+
+    row = q_sheet.max_row
+    calculation_index = get_column_position("calculation", q_sheet)
+
+    # Insert data as calculation from campaigns
+    if campaign_id and request.user.has_perm("iaso_polio") and polio_plugin:
+        get_data_from_campaigns(campaign_id, row, q_sheet, calculation_index)
+
+    filename = f"FORM_{form_name}_{datetime.now().date()}.xlsx"
+
+    with NamedTemporaryFile() as tmp:
+        wb.save(tmp.name)
+        tmp.seek(0)
+        stream = tmp.read()
+
+    return stream, filename
+
+
+def get_data_from_campaigns(campaign_id: str, row: int, q_sheet: Worksheet, calculation_index: int):
+    """This function get the data from a campaign and insert it in the calculation row
+    of the xlsform. It's tagged as 'insert_field' in the xlsform.
+    """
+    authorized_fields = [
+        "id",
+        "epid",
```

Comment:
Old PR I know, but I don't get why we still list the authorized fields if we want to pull the data from the api before sending it back to the generation API